lms/djangoapps/grades/api/v0/views.py

```diff
@@ -0,0 +1,142 @@
+from rest_framework.generics import GenericAPIView
+from rest_framework.permissions import IsAuthenticated
+from rest_framework.response import Response
+
+from lms.djangoapps.ccx.utils import prep_course_for_grading
+from lms.djangoapps.courseware import courses
+from lms.djangoapps.grades.new.course_grade import CourseGradeFactory
+from openedx.core.lib.api.authentication import OAuth2AuthenticationAllowInactiveUser
+
+log = logging.getLogger(__name__)
+
+
+class CurrentGradeView(GenericAPIView):
+    """
+    **Use Case**
+
+        * Get the current grade for a specific user in a course.
+
+    **Example Request**
+
+        GET /api/grades/v0/course_grade/{course_id}/?username={username}
+
+    **GET Parameters**
+
+        A GET request must include the following parameters.
+
+        * course_id: A string representation of a Course ID.
+        * username: A string representation of an user's username.
+
+    **GET Response Values**
+
+        If the request for information about the course progress status
+        is successful, an HTTP 200 "OK" response is returned.
+
+        The HTTP 200 response has the following values.
+
+        * username: A string representation of an user's username passed in the request.
+
+        * course_id: A string representation of a Course ID.
+
+        * passed: Boolean representing whether the course has been passed
+
+        * percent: A float represent the overall grade for the course
+
+        * grade: A letter grade as defined in grading_policy (e.g. 'A' 'B' 'C' for 6.002x) or None
+
+
+    **Example GET Response**
+
+        {
+            "username": "bob",
+            "course_key": "edX/DemoX/Demo_Course",
+            "passed": false,
+            "percent": 0.03,
+            "grade": None,
+        }
+
+    """
+    authentication_classes = (
+        JwtAuthentication,
+        OAuth2AuthenticationAllowInactiveUser,
+        SessionAuthentication,
+    )
+    permission_classes = (IsAuthenticated, )
+
+    def get(self, request, course_id):
+        """
+        Gets a course progress status.
+
+        Args:
+            request (Request): Django request object.
+            course_id (string): URI element specifying the course location.
+
+        Return:
+            A JSON serialized representation of the student current grade status.
+        """
+        username = request.GET.get('username')
+
+        # only the student can access her own grade status info
+        if request.user.username != username:
+            log.info('User %s tried to access user %s grade', request.user.username, username)
+            return Response(
+                status=status.HTTP_404_NOT_FOUND,
+                data={
+                    'error_code': 'user_does_not_exist'
+                }
+            )
+
+        # build the course key
+        try:
+            course_key = CourseKey.from_string(course_id)
+        except InvalidKeyError:
```

Comment:
`InvalidKeyError` does not imply that the course does not exist.  Rather, it means the provided course_id is invalid.  So instead of 404, return 400 with error_code "invalid_course_key".